### PR TITLE
feat: adds monitor_from_point method

### LINKED ISSUE #3057: [feat] `Monitor::from_point` and `Monitor::cursor_position`
### **Describe the bug**
I have two available monitor, and  I want to display the window on the current display，but the window.current_monitor() method does not seem to return the display screen where the mouse is located
It seems that the form always appears in the main display！
this is my code：
```
#[tauri::command]
fn register(app_handle: AppHandle, window: Window) {
  app_handle
    .global_shortcut_manager()
    .register("CTRL + U", move || {
      if window.is_visible().unwrap() {
        window.hide().unwrap();
      } else {
        println!("New version available: {:?}", window.available_monitors());
        window.center().unwrap();
        window.show().unwrap();
        window.set_focus().unwrap();
      }
    })
    .unwrap();
}
```
### **Platform and Versions (required):**

```
➜  power-clip git:(master) ✗ tauri info
 app:spawn Running "/Users/ocean/Documents/workplace/mystudywork/rust/github/tauri/tooling/cli.js/bin/tauri-cli tauri info" +0ms


Operating System - Mac OS, version 11.6.1 X64

Node.js environment
  Node.js - 16.13.0
  @tauri-apps/cli - 1.0.0-beta.10
  @tauri-apps/api - 1.0.0-beta.8

Global packages
  npm - 8.1.0
  yarn - 1.22.17

Rust environment
  rustc - 1.57.0
  cargo - 1.57.0

App directory structure
/dist
/target
/node_modules
/src-tauri
/.git
/.idea
/src

App

  tauri.rs - 1.0.0-beta.8
  build-type - bundle
  CSP - default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self' img-src: 'self'
  distDir - ../dist
  devPath - http://localhost:10001/
  bundler - Rollup
```

### **Additional context**
this is my monitors
```
Ok([Monitor { name: Some("Monitor #41012"), size: PhysicalSize { width: 2880, height: 1800 }, position: PhysicalPosition { x: 0, y: 0 }, scale_factor: 2.0 }, Monitor { name: Some("Monitor #30471"), size: PhysicalSize { width: 3840, height: 2160 }, position: PhysicalPosition { x: 0, y: -2160 }, scale_factor: 2.0 }])
```

### PR DESCRIPTION
**Pull request by clearlysid**
Exposes this [Tao method](https://docs.rs/tao/latest/tao/window/struct.Window.html#method.monitor_from_point) in Tauri too. Merging this closes #3057 and unblocks https://github.com/spyglass-search/spyglass/issues/312

### CODE REVIEW

File: core/tauri-runtime-wry/src/lib.rs [hunk 1]
```diff
@@ -1578,6 +1579,14 @@ impl<T: UserEvent> WindowDispatch<T> for WryWindowDispatcher<T> {
     Ok(window_getter!(self, WindowMessage::PrimaryMonitor)?.map(|m| MonitorHandleWrapper(m).into()))
   }
 
+  fn monitor_from_point(&self, x: f64, y: f64) -> Result<Option<Monitor>> {
+    // TODO: how do I pass x and y?
+    Ok(
+      window_getter!(self, WindowMessage::MonitorFromPoint)?
+        .map(|m| MonitorHandleWrapper(m).into()),
+    )
+  }
```
Comment (clearlysid) [inline core/tauri-runtime-wry/src/lib.rs:1595]: I think I'll need to pass on x and y to the WindowMessage but I'm not sure how? Adding it as arguments just gives me this error 🤔
Screenshot 2024-05-14 at 7 16 47 AM

Comment (clearlysid) [reply core/tauri-runtime-wry/src/lib.rs:1595]: It expects the `Sender<...>` type as first argument, but seems to get it by default even when no arguments are passed, so how come it demands that to be explicitly passed when I give additionally give it x and y? 😂

File: core/tauri/src/window/plugin.rs
Comment (clearlysid) [inline core/tauri/src/window/plugin.rs:1]: Similar situation here: I'm not sure how to "accept" x and y. None of the other getters need any args so I wonder if we need to handle this one differently?

**Comment by clearlysid** [comment]
If someone can guide me to resolve the above comments, this PR will be ready to review 💫

File: core/tauri/src/window/plugin.rs
Comment (amrbashir) [reply core/tauri/src/window/plugin.rs:1]: you'll need to create a command without the `getter!` macro, see `set_icon` or `toggle_maximize` below

File: core/tauri-runtime-wry/src/lib.rs (hunk 1 above)
Comment (amrbashir) [reply core/tauri-runtime-wry/src/lib.rs:1595]: you will have to use a manual approach here 

```rs
 	let (tx, rx) = channel();
 	send_user_message(
      &self.context,
      Message::Window(
        self.window_id,
        WindowMessage::MonitorFromPoint(tx, x, y),
      ),
    )
    
    rx
      .recv()
      .map_err(|_| crate::Error::FailedToReceiveMessage)
```

Comment (clearlysid) [reply core/tauri-runtime-wry/src/lib.rs:1595]: Thanks, this helped! I think I've approached it correctly now.

File: core/tauri/src/window/plugin.rs
Comment (clearlysid) [reply core/tauri/src/window/plugin.rs:1]: Updated this to not use the `getter!` macro in the last few commits.

File: core/tauri-runtime/src/lib.rs [hunk 2]
```diff
@@ -587,6 +589,11 @@ pub trait WindowDispatch<T: UserEvent>: Debug + Clone + Send + Sync + Sized + 's
   /// Returns None if it can't identify any monitor as a primary one.
   fn primary_monitor(&self) -> Result<Option<Monitor>>;
 
+  /// Find out the monitor on which the given point is located.
+  ///
+  /// Returns None if current monitor can't be detected.
```
Comment (amrbashir) [inline core/tauri-runtime/src/lib.rs]:
```suggestion
  /// Returns the monitor that contains the given point.
```

File: core/tauri/src/app.rs [hunk 3]
```diff
@@ -592,6 +592,17 @@ macro_rules! shared_app_impl {
         })
       }
 
+      /// Returns the monitor on which the given point exists.
+      ///
+      /// Returns None if it can't identify any monitor as a primary one.
```
Comment (amrbashir) [inline core/tauri/src/app.rs]:
```suggestion
  /// Returns the monitor that contains the given point.
```

File: core/tauri/src/webview/webview_window.rs [hunk 4]
```diff
@@ -1160,6 +1160,13 @@ impl<R: Runtime> WebviewWindow<R> {
     self.webview.window().primary_monitor()
   }
 
+  /// Returns the monitor on which the given point is.
+  ///
+  /// Returns None if monitor can't be detected.
```
Comment (amrbashir) [inline core/tauri/src/webview/webview_window.rs]:
```suggestion
  /// Returns the monitor that contains the given point.
```

File: core/tauri/src/window/mod.rs [hunk 5]
```diff
@@ -1437,6 +1437,18 @@ impl<R: Runtime> Window<R> {
       .map_err(Into::into)
   }
 
+  /// Returns the monitor on which the provided point is.
+  ///
+  /// Returns None if monitor can't be detected.
```
Comment (amrbashir) [inline core/tauri/src/window/mod.rs]:
```suggestion
  /// Returns the monitor that contains the given point.
```

File: tooling/api/src/window.ts [hunk 6]
```diff
@@ -2240,6 +2240,23 @@ async function primaryMonitor(): Promise<Monitor | null> {
   )
 }
 
+/**
+ * Returns the monitor on which the given point currently resides.
+ * Returns `null` if current monitor can't be detected.
```
Comment (amrbashir) [inline tooling/api/src/window.ts]:
```suggestion
  * Returns the monitor that contains the given point. Returns `null` if can't find any.
```

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thanks for contributing, LGTM, just need to get the documentation consistent and should be good to go. You also need to run `yarn build` and `yarn format` inside `tooling/api`

**Comment by clearlysid** [comment]
All good from my end ✅

**Review by amrbashir — DISMISSED** [review]
Thank you

File: core/tauri/src/app.rs [hunk 7]
```diff
@@ -592,6 +592,15 @@ macro_rules! shared_app_impl {
         })
       }
 
+  /// Returns the monitor that contains the given point.
+      pub fn monitor_from_point(&self, x: f64, y: f64) -> crate::Result<Option<Monitor>> {
```
Comment (amrbashir) [inline core/tauri/src/app.rs]:
```suggestion
      /// Returns the monitor that contains the given point.
      pub fn monitor_from_point(&self, x: f64, y: f64) -> crate::Result<Option<Monitor>> {
```

**Review by amrbashir — APPROVED** [review]

**Review by amrbashir — APPROVED** [review]

**Comment by clearlysid** [comment]
wow how'd you do this? here I thought I needed to go back and re-commit each one again with signature 😂 — nice work!

**Comment by amrbashir** [comment]
Check this for the command I use https://superuser.com/questions/397149/can-you-gpg-sign-old-commits/1123928#1123928

**Pull request merged**